src/crypto_utils.ts

```diff
@@ -144,27 +145,49 @@ export function buildInputText(
 
     const signature = signInputText(sender, contractAddress, functionSelector, ctInt);
 
-    return {ctInt, signature}
+    return {
+        ciphertext: ctInt,
+        signature: signature
+    }
 }
 
-export async function buildStringInputText(
+export function buildStringInputText(
     plaintext: string,
     sender: { wallet: BaseWallet; userKey: string },
     contractAddress: string,
     functionSelector: string
 ) {
     let encoder = new TextEncoder()
 
+    // Encode the plaintext string into bytes (UTF-8 encoded)        
     let encodedStr = encoder.encode(plaintext)
 
-    let encryptedStr = new Array<{ ciphertext: bigint, signature: Uint8Array }>(plaintext.length)
+    const inputText = {
+        ciphertext: { value: new Array<bigint> },
+        signature: new Array<Uint8Array>
+    }
+
+    // Process the encoded string in chunks of 8 bytes
+    // We use 8 bytes since we will use ctUint64 to store
+    // each chunk of 8 characters
+    for (let i = 0; i < encodedStr.length; i += EIGHT_BYTES) {
+        const startIdx = i
```

Comment:
if i is startIdx so why not just use startIdx instead of i

Reply:
Good idea